File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/03_Vacation.js
function vacation(input) {  // (["2000","1000","spend","1200","save","2000"]);
    let moneyNeeded = Number(input[0]);
    let moneyReal = Number(input[1]);

    
    let consecutiveSpendingDays = 0;
    let totalDays = 0;
    
    let i = 2;
    while (true) {
        let currentAction = input[i];
        i++;
        let currentAmount = Number(input[i]);
        i++;

        totalDays++;

        if (currentAction == "spend") {
            consecutiveSpendingDays++;
            moneyReal -= currentAmount;
            if (moneyReal < 0) {
                moneyReal = 0;
            }
        } else if (currentAction == "save") {
            moneyReal += currentAmount;
            consecutiveSpendingDays = 0;
        }

        if (consecutiveSpendingDays == 5) {
            console.log("You can't save the money.");
            console.log(totalDays);
            break;
        }

        if (moneyReal >= moneyNeeded) {
            console.log(`You saved the money for ${totalDays} days.`);
            break;
        }

    }
}


vacation([
    "2000",
    "1000",
    "spend",
    "1200",
    "save",
    "2000"
]);

File: Python_Courses/Python_Basics/Preparation/05_While_Loop_Exercise/03_Vacation.py
CONSECUTIVE_SPENDING_THRESHOLD = 5

total_days = 0
consecutive_spending = 0
is_failed = False

vacation_cost = float(input())
starting_money = float(input())

total_money = starting_money
while True:
    action = input()  # "spend" или "save"
    action_money = float(input())

    total_days += 1
    if action == "spend":
        consecutive_spending += 1
        if consecutive_spending >= CONSECUTIVE_SPENDING_THRESHOLD:
            is_failed = True
            break

        total_money -= action_money
        if total_money <= 0:
            total_money = 0

    elif action == "save":
        consecutive_spending = 0
        total_money += action_money

        if total_money >= vacation_cost:
            break


if not is_failed:
    print(f"You saved the money for {total_days} days.")
else:
    print("You can't save the money.")
    print(total_days)
